realsense2_camera/test/live_camera/rosci.py

```diff
@@ -0,0 +1,144 @@
+    print( "    Will find all tests and print" )
+
+    sys.exit( 2 )
+
+def command(dev_name):
+       cmd =  ['pytest-3']
+       cmd += ['-s']
+       cmd += ['-m', ''.join(dev_name)]
+       cmd += ['-k', 'test_camera_imu_tests'] 
+       #cmd += ['-m', 'd415']
+       cmd += [''.join(dir_live_tests)]
+       cmd += ['--debug']
+       return cmd
+
+def run_test(cmd, dev_name, stdout=None, append =False):
+        handle = None
+        try:
+            stdout = stdout + os.sep + str(dev_name) + '_' + "test_camera_imu_tests" + ".log"
+            if stdout is None:
+                sys.stdout.flush()
+            elif stdout and stdout != subprocess.PIPE:
+                if append:
+                    handle = open( stdout, "a" )
+                    handle.write(
+                        "\n----------TEST-SEPARATOR----------\n\n" )
+                    handle.flush()
+                else:
+                    handle = open( stdout, "w" )
+
+            result = subprocess.run( cmd,
+                    stdout=handle,
+                    stderr=subprocess.STDOUT,
+                    universal_newlines=True,
+                    timeout=200,
+                    check=True )
+
+        except Exception as e:
+             log.w( "Error Exception:\n ",e )
+                         
+        finally:
+            if handle:
+                handle.close()      
+
+device_set = None
+try:
+    opts, args = getopt.getopt( sys.argv[1:], 'hvqr:st:',
+                                longopts=['help', 'verbose', 'debug', 'quiet', 'regex=', 'stdout', 'tag=', 'list-tags',
+                                          'list-tests', 'no-exceptions', 'context=', 'repeat=', 'config=', 'no-reset', 'hub-reset',
+                                          'rslog', 'skip-disconnected', 'live', 'not-live', 'device='] )
+except getopt.GetoptError as err:
+    log.e( err )  # something like "option -a not recognized"
+    usage()
+for opt, arg in opts:
+    if opt in ('-h', '--help'):
+        usage()
+    elif opt == '--device':
+        device_set = arg.split()
+        for dev in device_set:
+             print(dev)        
+#Get into action    
+try:
+    logdir = os.path.join( os.path.dirname( os.path.dirname( os.path.dirname( os.path.abspath( __file__ )))), 'log')
+    os.makedirs( logdir, exist_ok=True )
+    print(logdir)
+    
+   
+
+    #Import '_device_by_sn' from devices.py module of librealsense repo
+    from rspy import devices
+    if not devices.hub:
+         assert False, 'hub not available'
+    else:
+        devices.hub.connect()
+    devices.query( hub_reset = hub_reset )
+    #devices.map_unknown_ports()
+    global _device_by_sn         
+
+    if not devices._device_by_sn:
+         assert False, 'No Camera Devices detected!'
+    else:
+         #Loop in for all devices and run tests
+         for device in devices._device_by_sn.values():
+              print(f'\033[93m Device found: {device.name} \033[0]')
```

Comment:
Device power cycle will come in another PR?

Reply:
Power cycle not considered for this PR now. I just disconnect the device after end of the test/CI.